File: Stripe/STPPaymentMethodSofortParams.m
//
//  STPPaymentMethodSofortParams.m
//  Stripe
//
//  Created by David Estes on 8/7/20.
//  Copyright © 2020 Stripe, Inc. All rights reserved.
//

#import "STPPaymentMethodSofortParams.h"

NS_ASSUME_NONNULL_BEGIN

@implementation STPPaymentMethodSofortParams

@synthesize additionalAPIParameters = _additionalAPIParameters;

+ (nullable NSString *)rootObjectName {
    return @"sofort";
}

+ (NSDictionary *)propertyNamesToFormFieldNamesMapping {
    return @{
                NSStringFromSelector(@selector(country)): @"country",
             };
}

@end

NS_ASSUME_NONNULL_END

File: StripePayments/StripePayments/Source/API Bindings/Models/PaymentMethods/Types/STPPaymentMethodSofortParams.swift
//
//  STPPaymentMethodSofortParams.swift
//  StripePayments
//
//  Created by David Estes on 8/7/20.
//  Copyright © 2020 Stripe, Inc. All rights reserved.
//

import Foundation

/// An object representing parameters used to create a Sofort Payment Method
public class STPPaymentMethodSofortParams: NSObject, STPFormEncodable {
    @objc public var additionalAPIParameters: [AnyHashable: Any] = [:]

    /// Two-letter ISO code representing the country the bank account is located in. Required.
    @objc public var country: String?

    @objc
    public class func rootObjectName() -> String? {
        return "sofort"
    }

    @objc
    public class func propertyNamesToFormFieldNamesMapping() -> [String: String] {
        return [
            NSStringFromSelector(#selector(getter: country)): "country"
        ]
    }
}
